pkg/types/aws_resource_manager.go

```diff
@@ -85,6 +85,25 @@ type AWSResourceManager interface {
 	// If the AWSResource does not support tags, only then the controller tags
 	// will not be added to the AWSResource.
 	EnsureTags(context.Context, AWSResource, ServiceControllerMetadata) error
+	// MirrorAWSTags ensures that AWS tags are included in the desired resource
+	// if they are present in the latest resource. This will ensure that the
+	// aws tags are not present in a diff. The logic of the controller will
+	// ensure these tags aren't patched to the resource in the cluster, and
+	// will only be present to make sure we don't try to remove these tags.
+	//
+	// Although there are a lot of similarities between this function and
+	// EnsureTags, they are very much different.
+	// While EnsureTags tries to make sure the resource contains the controller
+	// tags, mirrowAWSTags tries to make sure tags injected by AWS are mirrored
+	// from the latest resoruce to the desired resource.
+	MirrorAWSTags(resA AWSResource, resB AWSResource)
+	// FilterAWSTags ignores tags that have keys that start with "aws:"
+	// is needed to ensure the controller does not attempt to remove
+	// tags set by AWS. This function needs to be called after each Read
+	// operation.
+	// Eg. resources created with cloudformation have tags that cannot be
+	//removed by an ACK controller
+	FilterAWSTags(resA AWSResource)
```

Comment:
is this function intended to be called for desired & latest or only one of them? should we document that?